circuit/src/gadgets/set.rs

```diff
@@ -53,15 +64,23 @@ impl<F: FieldExt, const N: usize> FixedSetChip<F, N> {
 			let s_exp = v_cells.query_selector(s);
 
 			vec![
+				// If the difference is equal to 0, that will make the next product equal to 0.
+				// Example:
+				// product_exp = 1
+				// diff_exp = 0
+				// Check the constraint (1 * 0 == next_product_exp)
+				// That makes next_product_exp = 0
+				// => (1 * 0) - 0 == 0
 				s_exp.clone() * (product_exp * diff_exp.clone() - next_product_exp),
-				// TODO: uncomment this line when the bug is fixed.
-				// s_exp * (target_exp.clone() - (diff_exp + item_exp)),
+				//TODO: uncomment this line when the bug is fixed.
+				//s_exp * (target_exp - (diff_exp + item_exp)),
```

Comment:
We can fix this another time, since the errors are happening in the protocol crate